zulipterminal/ui_tools/boxes.py

```diff
@@ -502,6 +502,9 @@ def soup2markup(self, soup: Any, **state: Any) -> List[Any]:
                     link = urljoin(self.model.server_url, link)
 
                 text = text if text else link
+                # Only use the last segment if the text is a link itself.
+                if text == link:
+                    text = text.rstrip('/').split('/')[-1]
```

Comment:
We could consider parsing the URL using the standard library, as we do in other places.

Reply:
Resolved in PM. (We're moving ahead without the library call for now due to few inconsistencies. Though, we may want to switch to it later.)